### PR TITLE
[#943] Simplify Changing Author Order

### PR DESCRIPTION
**Pull request by mauriel133**
This PR implements dragging and dropping table rows in the Authors listing to change the order in which they appear.

image

### CODE REVIEW

**Comment by pkdash** [comment]
@hyi Can you review this?

File: theme/templates/resource-landing-page/modals.html [hunk 1]
```diff
@@ -356,17 +356,14 @@ <h4 class="modal-title" id="edit-creator"> Edit Author </h4>
...
-                                        id="id_creator-{{ author.order|add:"- 1" }}-order" name="creator-{{ author.order|add:"- 1" }}-order" required="required"
-                                        type="number" value="{% if author.order %}{{ author.order }}{% endif %}"></div>
-                            </div>
                         </fieldset>
                     </div>
                 </div>
+
+                {# Hidden in replacement for drag and drop, but still needed for form submission here #}
+                <input class="hidden-form" id="id_creator-{{ author.order|add:"- 1" }}-order"
+                       name="creator-{{ author.order|add:"- 1" }}-order" value="">
```
Comment (hyi) [inline theme/templates/resource-landing-page/modals.html]: Is there supposed to be a space between ```-``` and ```1``` here as a parameter?

File: hs_core/templates/pages/genericresource.html [hunk 2]
```diff
@@ -1424,6 +1471,36 @@ <h4 class="modal-title">Confirm files deletion</h4>
...
+
+                    $.ajax({
+                        type: "POST",
+                        url: url,
+                        data: datastring,
+                        success: function (result) {
+                            $("html").css("cursor", "initial");
+                        },
+                        error: function (xhr, errmsg, err) {
+                            $("html").css("cursor", "initial");
```
Comment (hyi) [inline hs_core/templates/pages/genericresource.html]: It'd be better to print the error message to console if ajax response returned with error message.

**Comment by hyi** [comment]
@mauriel133 I left a couple of minor comments. After I receive your response, I can +1.

File: theme/templates/resource-landing-page/modals.html (hunk 1 above)
Comment (mauriel133) [inline theme/templates/resource-landing-page/modals.html]: The space was not necessary. I removed it here and other places in this file.

File: hs_core/templates/pages/genericresource.html (hunk 2 above)
Comment (mauriel133) [inline hs_core/templates/pages/genericresource.html]: Agree, I will add it.

**Comment by mauriel133** [comment]
@hyi, I applied the suggested changes.

**Comment by hyi** [comment]
@mauriel133 +1 - you can merge it after it passes Jenkins check.

**Pull request merged**